### PR TITLE
[ASTGen/Macros] Adjustments to split `FunctionParameterSyntax` into multiple nodes for function parameters, closure parameters and enum parameters

### PR DESCRIPTION
**Pull request by ahoppen**
Companion of https://github.com/apple/swift-syntax/pull/1455

### CODE REVIEW

File: lib/ASTGen/Sources/ASTGen/Decls.swift [hunk 1]
```diff
@@ -89,11 +89,10 @@ extension ASTGenVisitor {
...
     let secondName: UnsafeMutableRawPointer?
-    let type: UnsafeMutableRawPointer?
-
-    if let nodeFirstName = node.firstName,
-       // Swift AST represnts "_" as nil.
-       nodeFirstName.text != "_" {
+    
+    let nodeFirstName = node.firstName
+    if nodeFirstName.text != "_" {
+      // Swift AST represnts "_" as nil.
```
Comment (kimdv) [inline lib/ASTGen/Sources/ASTGen/Decls.swift:95]: Is there a typo in `represnts`?

Comment (ahoppen) [reply lib/ASTGen/Sources/ASTGen/Decls.swift]: Possibly 😉

**Comment by ahoppen** [comment]
https://github.com/apple/swift-syntax/pull
https://github.com/apple/swift-format/pull/495
https://github.com/apple/swift-stress-tester/pull/229

@swift-ci Please smoke test

**Comment by ahoppen** [comment]
https://github.com/apple/swift-syntax/pull
https://github.com/apple/swift-format/pull/495
https://github.com/apple/swift-stress-tester/pull/229

@swift-ci Please smoke test macOS

**Comment by ahoppen** [comment]
https://github.com/apple/swift-syntax/pull/1455
https://github.com/apple/swift-format/pull/495
https://github.com/apple/swift-stress-tester/pull/229

@swift-ci Please smoke test macOS

**Pull request merged**